From 759817608d56aedd2b57e27939ad19d0c865d7ae Mon Sep 17 00:00:00 2001
From: Stefan Lange <s.lange@gateware.de>
Date: Wed, 12 Oct 2016 13:17:21 +0200
Subject: [PATCH] TQMT1042: Set MAC GTX Clock to be always provided by
 EC2_GTX_CLK125

As in some STKTxxxx SerDes configurations EC1 is inactive,
the MAC GTX clock is always provided by EC2 (EC2_GTX_CLK125).
---
 board/tqc/tqmt1042/tqmt1042.c | 11 +++++++++++
 include/configs/TQMT1042.h    |  2 ++
 2 files changed, 13 insertions(+)

diff --git a/board/tqc/tqmt1042/tqmt1042.c b/board/tqc/tqmt1042/tqmt1042.c
index d55f82b..9bf438e 100644
--- a/board/tqc/tqmt1042/tqmt1042.c
+++ b/board/tqc/tqmt1042/tqmt1042.c
@@ -241,6 +241,17 @@ int board_early_init_r(void)
 
 int misc_init_r(void)
 {
+/* MAC GTX Clock provided by EC2_GTX_CLK125
+ * As in some STKTxxxx SerDes configurations EC1 is inactive,
+ * GTX clock is always provided by EC2
+ */
+#ifdef CONFIG_SYS_TQMT1042_SCFG_EMIIOCR_EC2_GTX_CLK125
+        u32 emiiocr;
+        struct ccsr_scfg __iomem *scfg = (void *)CONFIG_SYS_MPC85xx_SCFG;
+
+        emiiocr = (in_be32(&scfg->emiiocr));
+        out_be32(&scfg->emiiocr, emiiocr |= 0x08000000);
+#endif
 	return 0;
 }
 
diff --git a/include/configs/TQMT1042.h b/include/configs/TQMT1042.h
index 5442a0a..eaabaa2 100644
--- a/include/configs/TQMT1042.h
+++ b/include/configs/TQMT1042.h
@@ -588,6 +588,8 @@
 #define CONFIG_SYS_TQMT1042_PLL1_CLKGEN_AUTOREPROG		/* automatic reprogramming of 849N202 clock generator based on RCW */
 #define CONFIG_SYS_849N202_I2C_BUSNR 0x1	/* I2C bus number where 849N202's is located */
 #define CONFIG_SYS_849N202_CLKGEN_ADDR 0x6C	/* 849N202's I2C address */
+/* MAC GTX Clock provided by EC2_GTX_CLK125 */
+#define CONFIG_SYS_TQMT1042_SCFG_EMIIOCR_EC2_GTX_CLK125
 
 /*
  * For booting Linux, the board info and command line data
-- 
1.9.1

